qiskit/transpiler/_basepass.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""This module implements the base pass.
+"""
+from abc import ABC, abstractmethod
+
+
+class BasePass(ABC):
+    """Base class for transpiler passes."""
+    @abstractmethod
+    def __init__(self):
+        """Base class for passes.
+
+        This method should initialize the module and its configuration, and
+        raise an exception if a component of the module is
+        not available.
+
+        Raises:
+            FileNotFoundError: if referenced passes are not registered.
+        """
+        # list of other passes required before this pass can run. this instructs the
```

Comment:
i made it `pass` for now because i'm removing all references to `requires` and `preserves`.

but once those come back, the `BasePass` should actually initialize the `requires` and `preserves` properties of all passes to None. This is a fundamental property of all passes, and cannot be left uninitialized.

Reply:
Great! If that's the case, we will come out with some way of enforcing it.